com.unity.render-pipelines.core/Editor/Lighting/LightUI.Drawers.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+namespace UnityEditor.Rendering
+{
+    /// <summary>
+    /// Contains a set of methods to help render the inspectors of Lights across SRP's
+    /// </summary>
+    public partial class LightUI
+    {
+        /// <summary>
+        /// Draws the color UI for a given light
+        /// </summary>
+        /// <param name="serialized"><see cref="ISerializedLight"/>The light to draw the color UI</param>
+        /// <param name="owner"><see cref="Editor"/>The editor that is handling the light</param>
+        public static void DrawColor(ISerializedLight serialized, Editor owner)
+        {
+            using (var changes = new EditorGUI.ChangeCheckScope())
```

Comment:
var changes are not used and can be removed

Reply:
Nice, I didn't know you can omit the variable assignment. Apparently compiler introduces a hidden variable in this case so the using block still works as expected. Good to know!

Reply:
Question though: As it is not used, do we need the using line at all?

Reply:
The using-scope ensures that the finalizer is called at the end of the scope (when the compiler-generated hidden variable falls out of scope), so it's still needed I think.

Reply:
AFAIK the finalizer of this scope only call EndChangeCheck() as `changes.changed` was never called. Also the balanceness of Begin-End is ensured by the scope mechanism but also checked in case of.
So here it is the same than doing:
```
BeginChangeCheck();
//all instruction in the using
if (EndChangeCheck())
    ; //nothing to do
```
And I don't get the point to do that.

Reply:
You might be right, but it's not obvious to me if it would be safe to remove the ChangeCheckScope. What if some code inside of the scope touches `GUI.changed`? If the scope is removed, then that "changed status" leaks out of this function, correct? It maybe that I don't yet fully understand how this all works so I'd trust your judgement on this though :) I guess my argument is that since Begin/EndChangeCheck touch global state, you need to be careful to not unintentionally change behavior.

Reply:
It leaks anyway. The way it is done is a push/pop mechanisme similar to OpenGL push/pop transformation matrix.
So when you `BeginChangeCheck()`, it add a layer above existing data with `GUI.changed` at false.
And when you `EndChangeCheck()`, it `GUI.changed` from the layer you are poping but also pass the result this way:
> GUI.changed |= s_ChangedStack.Pop();

Reply:
Ah I see. In this case you are then most likely right and the change scope is not needed when the scope variable is unused.

Reply:
this was tackled in another PR